src/EditorFeatures/Core/Shared/Tagging/TagSources/ProducerPopulatedTagSource.cs

```diff
@@ -675,7 +686,7 @@ public override ITagSpanIntervalTree<TTag> GetTagIntervalTreeForBuffer(ITextBuff
 
                     // We shall synchronously compute tags.
                     var producedTags = ConvertToTagTree(
-                        _tagProducer.ProduceTagsAsync(spansToCompute, GetCaretPoint(), CancellationToken.None).WaitAndGetResult(CancellationToken.None));
+                        _tagProducer.ProduceTagsAsync(spansToCompute, GetCaretPoint(), cancellationToken).WaitAndGetResult(cancellationToken));
```

Comment:
Looks like you're properly threading the cancellationToken around. Did you verify that we ever call ThrowIfCancellationRequested in the outlining tag producer?